### PR TITLE
Add components to Vue Boilerplate

### PR DESCRIPTION
**Pull request by sknep**
[draft - not ready to merge]

My goals for this PR were to:

- Demonstrate **importing the new astro-static npm package** and setting up a Vue Webpack project
- Demonstrate a simple **full-window layout using the global status bar and the tree using minimal flexbox** and existing Astro css variables
- Demonstrate **defining both simple (boolean) and complex (object or array) properties** in Vue, and **binding or passing that data down** to child components
- Demonstrate how a **click handler/method can be assigned to an Astro component**
- Demonstrate **2-way data binding or a props down/events up** parent/child communication

I ran into some challenges which are described on the ticket in Jira: https://rocketcom.atlassian.net/browse/ASTRO-248?focusedCommentId=11997 and abridged here:
- the Astro components are not actually `reflect`ing prop changes in the DOM, which means there’s no way for vue to know when something in a child component has been selected/changed state/etc.  I was not able to get the theme toggle component to reflect any changed state back to the parent. I can set the properties of any component on mount or any time I want, but I cannot seem to ‘get’ any data from the child components as they are today. **Either we accept that the astro components are sort of one-way, or we will need to add events that notify the parent when a property has changed.**
- Using the global status bar, tree, and notification components together on the same page required some extra fiddly css jury-rigging with position, overflow, and z-index -- they don't play together by default. **My recommendation here would be to simplify the internal CSS within the components so that they can work by default**, rather than expect an implementer to have a deep knowledge of flexbox, overflow, and position hacks.

Additionally, I found that the **fonts didn't load properly from the astro-static npm package** because all the URLs are absolute. I strongly recommend (https://github.com/RocketCommunicationsInc/astro-components/pull/3#discussion_r440979975) changing the urls in that package to relative within the css (cc @im-cr ) unless there's an implementation solution I'm missing entirely.

I'd very, very much like to **know how Todd and Cortney are handling the asset imports** and any other data binding for the React (https://rocketcom.atlassian.net/browse/ASTRO-249) and Angular (https://rocketcom.atlassian.net/browse/ASTRO-247) tickets.

This PR **cannot be merged until the astro-static assets 'work' out of the box**, as I've had to modify the font URLs in the src within node_modules to prototype this, and that won't do for a release.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,54 +1,46 @@
...
   },
   "dependencies": {
-    "vue": "^2.6.10",
-    "@astrouxds/rux-button": "^4.0.0",
-    "@astrouxds/rux-clock": "^4.0.0",
+    "@astrouxds/rux-accordion": "^4.0.0-alpha.3",
+    "@astrouxds/rux-button": "^4.0.2",
+    "@astrouxds/rux-clock": "^4.0.2",
     "@astrouxds/rux-global-status-bar": "^4.0.0",
     "@astrouxds/rux-icon": "^4.0.0",
```
Comment (dmcalester) [inline package.json:14]: Was there a reason only come of the packages were updated to their most recent versions? I would recommend upgrading button, icon, monitoring icon and clock to reflect the most recent changes.

Comment (sknep) [reply package.json:14]: - [ ] Review each astro dependency and update accordingly

File: index.html [hunk 2]
```diff
@@ -2,10 +2,12 @@
...
     <meta charset="utf-8">
-    <title>Astro Vue Bootstrap</title>
+    <title>Astro Vue Boilerplate</title>
+    <link rel="icon" href="<%= BASE_URL %>favicon.ico">
+
   </head>
   <body>
     <div id="app"></div>
-    <script src="/dist/build.js"></script>
+        <!-- built files will be auto injected -->
```
Comment (amalya-rocket) [inline index.html:11]: Not sure if this is applicable here, but...

auto-injected

File: README.md [hunk 3]
```diff
@@ -1,18 +1,24 @@
 # Astro Vue Bootstrap
```
Comment (amalya-rocket) [inline README.md:1]: Should this be Boilerplate instead?

File: index.html (hunk 2 above)
Comment (amalya-rocket) [inline index.html:11]: Should this be auto-injected with a hyphen?

File: package.json [hunk 4]
```diff
@@ -1,54 +1,46 @@
 {
-  "name": "astro-vue-boilerplate",
-  "description": "Astro Vue Boilerplate",
-  "version": "1.0.0",
-  "author": "Rocket Communications",
-  "license": "MIT",
+  "name": "astro-vue",
```
Comment (amalya-rocket) [inline package.json:2]: Is there a reason not to include a description, author, or license any more? Or to remove the term Boilerplate from he name?

File: src/App.vue [hunk 5]
```diff
@@ -1,22 +1,180 @@
...
+    flex: 4 1 auto;
+}
+rux-global-status-bar {
+  position: relative; /* un-sticky GSB */
+}
+.rux-button-group {
+    margin-left: auto;
+    align-self: flex-end;
+}
+rux-accordion [slot='content'] {
```
Comment (amalya-rocket) [inline src/App.vue:176]: Is it okay that this line says content and the next one says contents (plural)?

**Pull request closed without merging**